src/ert/storage/local_ensemble.py

```diff
@@ -357,15 +353,24 @@ def load_gen_data_as_df(
         return pd.concat(dfs)
 
     def load_field(self, key: str, realizations: List[int]) -> npt.NDArray[np.double]:
-        result = []
+        result: Optional[npt.NDArray[np.double]] = None
         for realization in realizations:
             input_path = self.mount_point / f"realization-{realization}"
             if not input_path.exists():
                 raise KeyError(f"Unable to load FIELD for key: {key}")
             data = np.load(input_path / f"{key}.npy")
-            data_no_nans = data[~np.isnan(data)]
-            result.append(data_no_nans)
-        return np.stack(result).T  # type: ignore
+            data = data[np.isfinite(data)]
+
+            if result is None:
```

Comment:
Are we sure that all the fields produced will have the same elements be `NaN` or `inf`?
Perhaps add a check like the following?

```python
            if result is None:
                result = np.empty((len(realizations), data.size), dtype=np.double)
            elif data.size != result.shape[1]:
                raise ValueError(f"Data size mismatch for realization {realization}")
```

Reply:
we are sure that they will always have the same cells masked with NaNs as the code is now, but it doesn't hurt to add that test.